### PR TITLE
add ghc 9.10 support

### PR DESCRIPTION
**Pull request by rsoeldner**

### CODE REVIEW

**Comment by conal** [comment]
Thanks for the PR!

What was breaking in GHC 9.10, and why?

Under which GHC versions have you successfully compiled this PR?

**Comment by rsoeldner** [comment]
I tested it with `9.6.4`, `9.8.2`, and `9.10.1`.

`liftA2` is now exposed by `Prelude`: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.6#lifta2-exported-by-prelude

The additional constraints are required since 9.10 as described here: https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/9.10#loopy-superclasses

**Comment by conal** [comment]
Terrific. Thanks again!

File: src/Data/AffineSpace.hs [hunk 1]
```diff
@@ -21,8 +21,9 @@ module Data.AffineSpace
   (
     AffineSpace(..), (.-^), distanceSq, distance, alerp, affineCombo
   ) where
-
+#if !MIN_VERSION_base(4,10,0)
```
Comment (andreasabel) [inline src/Data/AffineSpace.hs:24]: That's bonkers.
`liftA2` only entered `Prelude` in base-4.18.

File: src/Data/VectorSpace.hs [hunk 2]
```diff
@@ -32,11 +32,13 @@ module Data.VectorSpace
   , InnerSpace(..)
   , lerp, linearCombo, magnitudeSq, magnitude, normalized, project
   ) where
-
+#if !(MIN_VERSION_base(4,8,0))
```
[the remark this answers is not in the document]
Comment (rsoeldner) [reply src/Data/VectorSpace.hs:35]: Indeed, this seems to be incorrect. I don't see why it should be 4.18 and not 4.10.

**Pull request merged**